src/controller/p2p.rs

```diff
@@ -1,9 +1,11 @@
-use std::{env, sync::Arc, time::Duration};
+use std::{env, io::Write, sync::{Arc, Mutex}, time::Duration};
```

Comment:
Please use `parking_lot::Mutex` instead of `std::Mutex`, it is much faster

Reply:
done